### PR TITLE
Fix of NuGet/Home#1543: Nuget 3.2.0.10516 command line client cannot …

### PR DESCRIPTION
**Pull request by feiling**
…restore packages under mono.

When running on Mono, nuget now uses the XBuild's SolutionParser to get the list of projects in a solution. Tested with Mono 4.0.5.1 on Ubuntu 14.04.

@yishaigalatzer @emgarten @deepakaravindr

### CODE REVIEW

File: src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs [hunk 1]
```diff
@@ -393,7 +459,6 @@ private static Version SafeParseVersion(string version)
             }
         }
 
-
```
Comment (feiling) [inline src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs:396]: Just format changes here (remove trailing spaces)

File: src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs [hunk 2]
```diff
@@ -124,7 +125,58 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
...
 
-        public static IEnumerable<string> GetAllProjectFileNames(string solutionFile, string msbuildPath)
+        /// <summary>
+        /// Gets the list of project files in a solution, using XBuild's solution parser.
+        /// </summary>
+        /// <param name="solutionFile">The solution file. </param>
+        /// <returns>The list of project files (in full path) in the solution.</returns>
+        public static IEnumerable<string> GetAllProjectFileNamesWithXBuild(string solutionFile)
+        {
+            try
```
Comment (feiling) [inline src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs:135]: This code is copies from 2.10

**Comment by yishaigalatzer** [comment]
:shipit:

**Comment by emgarten** [comment]
:rocket:

**Comment by feiling** [comment]
Fix merged

**Pull request closed without merging**